lightning/src/ln/channelmanager.rs

```diff
@@ -2548,6 +2546,40 @@ pub enum RecentPaymentDetails {
 	},
 }
 
+/// Error during creation and handling of BOLT 12 related payments.
```

Comment:
The roles of Bolt12CreationError, and Bolt12Response feels a little confusing to me.

<details>
<summary>
For example
</summary>

It reads _during creation and handling of BOLT 12 related payments_ but technically we are not creating any payment when we are creating an offer.

</details>


Maybe we can simplify the enums, by introducing only one new enum called `Bolt12CreationError` that contains "Error during creation of BOLT12 Messages"?